src/verifai/scenic_server.py

```diff
@@ -62,3 +73,160 @@ def _simulate(self, scene):
+        else:
+            self.rejectionFeedback = extSampler.rejectionFeedback
+        defaults = DotMap(maxSteps=None, verbosity=0, maxIterations=1)
+        defaults.update(options)
+        self.maxSteps = defaults.maxSteps
+        self.verbosity = defaults.verbosity
+        self.maxIterations = defaults.maxIterations
+
+    def get_sample(self, sample):
+        self.sampler.scenario.externalSampler.last_sample = sample
+        self.full_sample = self.sampler.nextSample(sample)
+
+    def simulate(self, sample):
+
+        '''
+        Need to generate scene from sample here.
+        '''
+        t0 = time.time()
+        self.sampler.scenario.externalSampler.last_sample = sample
+        scene = self.sampler.lastScene
+        startTime = time.time()
+        if self.verbosity >= 1:
+            print('  Beginning simulation...')
+        try:
+            result = self.simulator.simulate(scene,
+                maxSteps=self.maxSteps, verbosity=self.verbosity,
+                maxIterations=self.maxIterations)
+            result.worker_num = self.worker_num
+        except SimulationCreationError as e:
+            if self.verbosity >= 1:
+                print(f'  Failed to create simulation: {e}')
+            self.lastValue = self.rejectionFeedback
+            return self.worker_num, self.full_sample, self.lastValue
+        except RuntimeError as e:
+            print(f'Runtime error during simulation: {e}')
+            print('Waiting 1 minute before continuing...')
+            time.sleep(60)
+            return self.worker_num, self.full_sample, self.lastValue
+        if self.verbosity >= 1:
+            totalTime = time.time() - startTime
+            print(f'  Ran simulation in {totalTime:.4g} seconds.')
+        if result is None:
+            self.lastValue = self.rejectionFeedback
+        else:
+            self.lastValue = (0 if self.monitor is None
+                              else self.monitor.evaluate(result))
+        return self.worker_num, self.full_sample, self.lastValue
+
+class ParallelScenicServer(ScenicServer):
+
+    def __init__(self, total_workers, n_iters, sampling_data, scenic_path, monitor,
+    options={}, use_carla=False, max_time=None, scenario_params={}, sampler=None):
+        if not ray.is_initialized():
+            ray.init(ignore_reinit_error=True)
+        self.total_workers = total_workers
+        self.n_iters = n_iters
+        self.max_time = max_time
+        # sampler = ScenicSampler.fromScenario(scenic_path, **scenario_params)
+        sampling_data.sampler = sampler
+        super().__init__(sampling_data, monitor, options)
+        print(f'Sampler class is {type(self.sampler)}')
+        self.sample_simulators = [SampleSimulator.remote(scenic_path, i, monitor, options,
+        use_carla, scenario_params)
+        for i in range(self.total_workers)]
+
+    def _generate_next_sample(self, worker_num):
+        i = 0
+        feedback = self.lastValue
```

Comment:
Doesn't look like this is getting used any more (I guess the update is happening elsewhere?)